plugins/source/aws/client/multiplexers.go

```diff
@@ -12,15 +10,17 @@ var AllNamespaces = []string{ // this is only used in applicationautoscaling
 }
 
 // Extract region from service list
-func getRegion(regionalMap map[string]*Services) string {
-	if len(regionalMap) == 0 {
+func getRegion(regionalList []string) string {
+	// TODO: We should try and find the closest region if possible. This will require checking the following locations:
+	// 1. Region defined by the ec2 metadata service
+	// 2. Region defined by the AWS_REGION environment variable
+	// 3. Region defined by the AWS_DEFAULT_REGION environment variable
+	// 4. Region defined by the local config file
+	if len(regionalList) == 0 {
 		return ""
 	}
-	regions := make([]string, 0)
-	for i := range regionalMap {
-		regions = append(regions, i)
-	}
-	sort.Strings(regions)
```

Comment:
What do you think about keeping the `sort.Strings(region)` in here to maintain consistency with the previous implementation? IIRC, we had it here to prevent some intermittent bugs caused by Go map ordering, but removing it now will change the order of regions again, so maybe some users will run into issues with the new ordering